### PR TITLE
[MXNET-1291] solve pylint errors in examples with issue no.12205

### PR DESCRIPTION
**Pull request by cchung100m**
## Description ##
this is the part 3 of the changes to solve pylint errors in examples with issue #12205 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: example/bayesian-methods/bdk_demo.py [hunk 1]
```diff
@@ -58,6 +62,9 @@ def backward(self, out_grad, in_data, out_data, in_grad):
 
 
 class LogSoftmax(mx.operator.NumpyOp):
+    """
+    Generate helper functions to evaluate softmax loss function
```
Comment (ChaiBapchya) [inline example/bayesian-methods/bdk_demo.py]: loss function? Shouldn't this be logarithm of softmax

Comment (ChaiBapchya) [reply example/bayesian-methods/bdk_demo.py]: nitpick : At times you have used "calculate" and at times "Generate helper functions to evaluate"
Should have consistency right?

File: example/bayesian-methods/bdk_demo.py [hunk 2]
```diff
@@ -103,6 +110,9 @@ def regression_student_grad(student_outputs, teacher_pred, teacher_noise_precisi
 
 
 def get_mnist_sym(output_op=None, num_hidden=400):
+    """
+    get symbol of mnist
```
Comment (ChaiBapchya) [inline example/bayesian-methods/bdk_demo.py]: nitpick : "G"et

File: example/caffe/caffe_net.py [hunk 3]
```diff
@@ -14,29 +14,37 @@
...
+import argparse
 import mxnet as mx
 from data import get_iterator
-import argparse
 import train_model
 
+
 def get_mlp():
     """
     multi-layer perceptron
```
Comment (ChaiBapchya) [inline example/caffe/caffe_net.py]: nitpick : Add Get to maintain consistency of function comment

File: example/caffe/train_model.py [hunk 4]
```diff
@@ -14,12 +14,18 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
-import mxnet as mx
-import logging
+"""
+Train module with using Caffe operator in MXNet
```
Comment (ChaiBapchya) [inline example/caffe/train_model.py]: not need for with

Train module using Caffe operator in MXNet

File: example/caffe/train_model.py [hunk 5]
```diff
@@ -14,12 +14,18 @@
...
+"""
 import os
+import logging
+import mxnet as mx
+
 
 def fit(args, network, data_loader, eval_metrics=None, batch_end_callback=None):
+    """
+    Train the model with using Caffe operator in MXNet
```
Comment (ChaiBapchya) [inline example/caffe/train_model.py]: same as above

File: example/capsnet/capsulenet.py [hunk 6]
```diff
@@ -14,24 +14,31 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-import mxnet as mx
-import numpy as np
+"""
+Generate MXNet implementation of CapsNet
```
Comment (ChaiBapchya) [inline example/capsnet/capsulenet.py]: Capsule Net
Can we add reference of the paper in comment?

File: example/capsnet/capsulenet.py [hunk 7]
```diff
@@ -184,6 +198,9 @@ def __call__(self, num_update):
 
 
 def do_training(num_epoch, optimizer, kvstore, learning_rate, model_prefix, decay):
+    """
+    Run training to CapsNet
```
Comment (ChaiBapchya) [inline example/capsnet/capsulenet.py]: How about "Perform CapsNet training"

File: example/capsnet/capsulenet.py [hunk 8]
```diff
@@ -255,30 +273,53 @@ def random_shift(x, width_shift_fraction, height_shift_fraction):
     x = apply_transform(x, shift_matrix, 'nearest')
     return x
 
+
 def _shuffle(data, idx):
     """Shuffle the data."""
```
Comment (ChaiBapchya) [inline example/capsnet/capsulenet.py]: At times single line comment are used and at times multi-line comment are used even if the actual comment is just one line. Need to establish one way out of the two

File: example/cnn_chinese_text_classification/data_helpers.py [hunk 9]
```diff
@@ -14,6 +14,9 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+
+"""Help functions to support for implementing CNN + Highway Network for Chinese Text Classification in MXNet"""
```
Comment (ChaiBapchya) [inline example/cnn_chinese_text_classification/data_helpers.py]: """
abc
"""
vs """ abc """

Comment (ChaiBapchya) [reply example/cnn_chinese_text_classification/data_helpers.py]: Another nitpick "Helper"

File: example/cnn_chinese_text_classification/data_helpers.py [hunk 10]
```diff
@@ -116,30 +124,30 @@ def build_vocab(sentences):
...
         x_vec.append(vec)
     x_vec = np.array(x_vec)
     y_vec = np.array(labels)
     return [x_vec, y_vec]
 
 
-def load_data_with_word2vec(word2vec):
+def load_data_with_word2vec(word2vec_list):
     """
     Loads and preprocessed data for the MR dataset.
```
Comment (ChaiBapchya) [inline example/cnn_chinese_text_classification/data_helpers.py]: preprocesses*

File: example/cnn_text_classification/data_helpers.py [hunk 11]
```diff
@@ -15,6 +15,8 @@
 # specific language governing permissions and limitations
 # under the License.
 
+"""Help functions to support for implementing CNN + Highway Network for Text Classification in MXNet"""
```
Comment (ChaiBapchya) [inline example/cnn_text_classification/data_helpers.py]: Helper

File: example/cnn_text_classification/data_helpers.py [hunk 12]
```diff
@@ -111,23 +113,24 @@ def build_input_data(sentences, labels, vocabulary):
...
         x_vec.append(vec)
     x_vec = np.array(x_vec)
     y_vec = np.array(labels)
     return [x_vec, y_vec]
 
 
-def load_data_with_word2vec(word2vec):
+def load_data_with_word2vec(word2vec_list):
     """
     Loads and preprocessed data for the MR dataset.
```
Comment (ChaiBapchya) [inline example/cnn_text_classification/data_helpers.py]: preprocesses

File: example/ctc/multiproc_data.py [hunk 13]
```diff
@@ -15,6 +15,8 @@
 # specific language governing permissions and limitations
 # under the License.
 
+"""Contains a class for handling multi-process data generation"""
```
Comment (ChaiBapchya) [inline example/ctc/multiproc_data.py]: """a""" vs 
"""
a
"""

File: example/gluon/dc_gan/dcgan.py [hunk 14]
```diff
@@ -128,9 +145,20 @@ def transformer(data, label):
 
 
 # get dataset with the batch_size num each time
-def get_dataset(dataset):
+def get_dataset(dataset_name):
+    """
+    Load the dataset and split dataset to train/valid data
```
Comment (ChaiBapchya) [inline example/gluon/dc_gan/dcgan.py]: nitpick split "it"

File: example/gluon/lstm_crf/lstm_crf.py [hunk 15]
```diff
@@ -15,44 +15,53 @@
 # specific language governing permissions and limitations
 # under the License.
 
+"""
+# This example demonstrates how the LSTM-CRF model can be implemented
```
Comment (ChaiBapchya) [inline example/gluon/lstm_crf/lstm_crf.py]: Why #?

File: example/capsnet/capsulenet.py (hunk 7 above)
Comment (cchung100m) [reply example/capsnet/capsulenet.py]: Agree, it is a better description.

File: example/capsnet/capsulenet.py (hunk 6 above)
Comment (cchung100m) [reply example/capsnet/capsulenet.py]: Copy that, it is good to help programmers to understand it.

File: example/capsnet/capsulenet.py (hunk 8 above)
Comment (cchung100m) [reply example/capsnet/capsulenet.py]: You are right, I think the style of comments should align to multi-line comment.

File: example/bayesian-methods/bdk_demo.py (hunk 1 above)
Comment (cchung100m) [reply example/bayesian-methods/bdk_demo.py]: Thanks for pointing out the misdescription of softmax, and I agree to use "calculate" here, it's a better description.

**Comment by cchung100m** [comment]
@ChaiBapchya ,
I solve the conflict of bdk_demo.py and re-trigger the CI. thank you for the suggestions. :)

**Comment by cchung100m** [comment]
Hi @marcoabreu 

Sorry, I don’t know what you want to tell me. Can you elaborate on that? 

Many thanks,

File: example/autoencoder/model.py [hunk 16]
```diff
@@ -53,7 +54,7 @@ def extract_feature(sym, args, auxs, data_iter, N, xpu=mx.cpu()):
 
 
 class MXModel(object):
-    def __init__(self, xpu=mx.cpu(), *args, **kwargs):
+    def __init__(self, *args, xpu=mx.cpu(), **kwargs):
```
Comment (sandeep-krishnamurthy) [inline example/autoencoder/model.py]: why?

File: example/autoencoder/solver.py [hunk 17]
```diff
@@ -19,9 +19,8 @@
 from __future__ import print_function
 
 import logging
-
-import mxnet as mx
 import numpy as np
+import mxnet as mx
```
Comment (sandeep-krishnamurthy) [inline example/autoencoder/solver.py]: Why mxnet import should be moved down?

File: example/bayesian-methods/algos.py [hunk 18]
```diff
@@ -14,13 +14,15 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
+"""
```
Comment (sandeep-krishnamurthy) [inline example/bayesian-methods/algos.py]: In MXNet we follow comments like
```
"""Your comment...
"""

File: example/gluon/audio/urban_sounds/README.md [hunk 19]
```diff
@@ -1,100 +0,0 @@
-# Urban Sounds Classification in MXNet Gluon
```
Comment (sandeep-krishnamurthy) [inline example/gluon/audio/urban_sounds/README.md]: Why is this removed?

**Comment by sandeep-krishnamurthy** [comment]
@cchung100m - Thank you for your contributions and solving pylint issues in examples. I have left few comments.
How are we testing that examples are not broken after this change? @marcoabreu

File: example/autoencoder/solver.py (hunk 17 above)
Comment (cchung100m) [reply example/autoencoder/solver.py]: Solve pylint error message such as [C0411 ] standard import "import numpy as np" should be placed before "import mxnet as mx"

**Comment by marcoabreu** [comment]
We don't test examples in CI :(

File: example/autoencoder/model.py (hunk 16 above)
Comment (cchung100m) [reply example/autoencoder/model.py]: I forget that why I change it, and it seems like the position of *args does not impact anything in the module, therefore I intend to keep the original design(*args parameter at 2nd).

File: example/gluon/audio/urban_sounds/README.md (hunk 19 above)
Comment (cchung100m) [reply example/gluon/audio/urban_sounds/README.md]: Sorry, I don't know anything about this file, I just accept this change which made by communities before committing my branch.

**Comment by vandanavk** [comment]
@cchung100m are the files in example/autoencoder being deleted?

@marcoabreu we might have to wait for the fixes to all the lint errors (in examples folder) to be merged, before we enable the check in CI

**Comment by ankkhedia** [comment]
@cchung100m Could you please look into the comments above?

**Comment by cchung100m** [comment]
@vandanavk 
Sorry for the late reply.
Yes, the files in example/autoencoder being deleted.

@ankkhedia 
Thanks for the reminder.

I will solve the conflict files and rebase this branch ASAP.

File: example/bayesian-methods/algos.py (hunk 18 above)
Comment (cchung100m) [reply example/bayesian-methods/algos.py]: thanks for reminding :)

**Comment by vandanavk** [comment]
@marcoabreu @ChaiBapchya @sandeep-krishnamurthy is this PR good to go?

**Comment by cchung100m** [comment]
I will re-check the conflicting files ASAP.

**Comment by cchung100m** [comment]
Hi @szha @marcoabreu @ChaiBapchya @sandeep-krishnamurthy 

I complete the rebase and [re-test](https://gist.github.com/cchung100m/85c357b4fbda76f93a939a050303940a) on this branch, I will appreciate if you can help to review/merge, again, thanks :)
